src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useRef, useState } from "react";
+import { useMediaQuery } from "react-responsive";
+import { useInView } from "react-intersection-observer";
+import { useTimer } from "./use-timer";
+import { CountdownTimerProps } from "./types";
+import { Countdown, Time, TimeLeft, Wrapper } from "./countdown-timer.style";
+import { TimeHelper } from "../util/time-helper";
+import { ClockIcon } from "@lifesg/react-icons";
+import { MediaWidths } from "../spec/media-spec";
+
+type Position = "relative" | "fixed";
+
+export const CountdownTimer = ({
+    className,
+    timer,
+    notifyTimer,
+    offset,
+    mobileOffset,
+    show,
+    "data-testid": testId,
+    onDuration,
+    onNotify,
+    onFinish,
+}: CountdownTimerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+
+    const wrapperRef = useRef<HTMLDivElement>();
+    const isNotified = useRef<boolean>(false);
+    const [offsetY, setOffsetY] = useState<number>(0);
+    const [isPlaying, setIsPlaying] = useState(false);
+
+    const [remainingSeconds] = useTimer(timer, isPlaying);
+    const { ref: stickyRef, inView } = useInView({
+        threshold: 1,
```

Comment:
there's a slight flickering at the start, I found that setting `initialInView: true` resolves this